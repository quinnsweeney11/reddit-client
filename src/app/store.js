import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import postContainerReducer from '../containers/postContainer/postContainerSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    postContainer: postContainerReducer
  },
});
