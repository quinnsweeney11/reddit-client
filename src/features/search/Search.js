import { setCurrentSub } from "./searchSlice";
import { useDispatch } from "react-redux";
import { useState } from "react";
import "./search.css";

export function Search(props) {
  const [state, setState] = useState("");
  const dispatch = useDispatch();
  const handleChange = (e) => {
    if (e.target.value[e.target.value.length - 1] !== " ") {
      setState(e.target.value.substring(2));
    }
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (state.length !== 0) {
      dispatch(setCurrentSub(state));
    } else {
      alert(`Please enter a value before searching.`);
    }
  };

  return (
    <div className="search">
      <form onSubmit={(e) => handleSubmit(e)} class="search-form">
        <input
          type="text"
          class="search-text"
          placeholder="subreddit"
          value={`r/${state}`}
          onChange={(e) => handleChange(e)}
        />
        <input
          type="button"
          class="search-button"
          value="Search"
          onClick={(e) => handleSubmit(e)}
        />
      </form>
    </div>
  );
}
