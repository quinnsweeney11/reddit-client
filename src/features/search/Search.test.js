import React from "react";
import { Search } from "./Search";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";

describe("snapshot", () => {
  it("test", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Search />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
