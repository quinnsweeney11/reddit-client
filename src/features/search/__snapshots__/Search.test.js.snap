// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="search"
>
  <form
    class="search-form"
    onSubmit={[Function]}
  >
    <input
      class="search-text"
      onChange={[Function]}
      placeholder="subreddit"
      type="text"
      value="r/"
    />
    <input
      class="search-button"
      onClick={[Function]}
      type="button"
      value="Search"
    />
  </form>
</div>
`;
